app/controllers/fhir/r4/api_controller.rb

```diff
@@ -208,11 +215,49 @@ def update
                                  comment: "Vaccination edited via the API (ID: #{vaccine.id}).")
       end
       status_ok(vaccine.as_fhir) && return
+    when 'observation'
+      return if doorkeeper_authorize!(*OBSERVATION_WRITE_SCOPES)
+
+      # Get the Lab that needs to be updated
+      lab = get_record(Laboratory, params.permit(:id)[:id])
+      status_forbidden && return if lab.nil?
+
+      # Get the contents from applying a patch, if needed
+      if request.patch? && !lab.nil?
+        begin
+          contents = apply_patch(lab, patch)
+        rescue StandardError => e
+          status_bad_request([['Unable to apply patch', e&.message].compact.join(': ')]) && return
+        end
+      end
+
+      fhir_map = laboratory_from_fhir(contents)
+      request_updates = fhir_map.transform_values { |v| v[:value] }
+      status_unprocessable_entity && return if request_updates.nil?
+
+      # Assign any remaining updates to the Lab
+      lab.assign_attributes(request_updates)
+
+      # Wrap updates to the Lab and History creation in a transaction
+      ActiveRecord::Base.transaction do
+        unless referenced_patient_valid_for_client?(lab, :patient_id) && lab.save(context: :api) && fhir_map.all? { |_k, v| v[:errors].blank? }
+          req_json = request.patch? ? lab.as_fhir.to_json : JSON.parse(request_body)
+          status_unprocessable_entity(lab, fhir_map, req_json) && (raise SaveError)
+        end
+
+        Rails.logger.info "Updated Lab Result (ID: #{lab.id}) for Patient with ID: #{lab.patient_id}"
+        History.lab_result_edit(patient: lab.patient_id,
+                                created_by: @current_actor_label,
+                                comment: "Lab Result edited via the API (ID: #{lab.id}).")
+      end
+      status_ok(lab.as_fhir) && return
     else
       status_not_found && return
     end
   rescue JSON::ParserError
     status_bad_request(['Invalid JSON in request body'])
+  rescue SaveError
+    nil
```

Comment:
Leave an inline comment explaining why you are doing this

Reply:
https://github.com/SaraAlert/SaraAlert/pull/870/commits/d27d21a2163d3c613b633349fd3eb75f5fdaccf8